lib/net/validations.rb

```diff
@@ -63,6 +63,21 @@ def self.validate_mac(mac)
       false
     end
 
+    def self.multicast_mac?(mac)
+      return false unless validate_mac(mac)
+
+      # Get most signitifcant byte
+      msb = mac.gsub(/[\.\:\-]/, '').slice(0..1).to_i(16)
+
+      # Is most significant bit set?
+      msb & 0b1 == 1
+    end
+
+    def self.broadcast_mac?(mac)
+      return false unless validate_mac(mac)
+      mac.gsub(/[\.\:\-]/, '').downcase == 'ffffffffffff'
```

Comment:
ditto, prefer `tr` over gsub - or even remove entirely, as `mac.downcase` can be compared to `ff:ff:ff:ff:ff:ff`